moviebuff.go

```diff
@@ -35,71 +35,95 @@ var (
 
 // Moviebuff allows to access to information in moviebuff using resource ids.
 
+type Logger interface {
```

Comment:
Let us ideally not log anything here.

Reply:
@maknahar : ^ Can we remove it? @sivachandran

Reply:
@sudarshan-reddy Logs are used for debugging while development. Need not be passed in production. That is the reason we have it as an interface. If we don;lt have this only way to debug is to do print manually and remove once done.

Reply:
Ideally all the errors should be wrapped with additional information and passed back to the caller.

Reply:
The existing SDKs like `moviebuff-sdk-go` and `qubeaccount-sdk-go` are implemented like this, Do I need to address it now? since I don't know, how it will affect in all caller places. @sivachandran  @maknahar @sudarshan-reddy

Reply:
@sivachandran @maknahar @sudarshan-reddy ^

Reply:
@maknahar Sorry for the late reply. Can we not use the logs when debugging ad-hoc? Seems like if we add unit tests, we might not have the need to log debugs all that much at all. I see no harm in having it, but it just adds sirupsen/logrus as an extra package.

Reply:
@Malathis Compatibility with other applications using this struct? If yes, then the answer is Yes and No. 

Vendor lock-in means they will not face any issues on breaking changes (An addition to a field may not be a breaking change. A change in the name of a field or it's json tag will definitely be one).

However, we need to always assume they may need to update their library and see if we can somehow make the changes compatible. If there is no chance at all, let us communicate that and document it.

Reply:
Okay @sudarshan-reddy, then I will remove that compatible changes which I have done on `qubeaccount-sdk-go` - https://github.com/RealImage/qube-account-sdk-go/pull/6/files#diff-6f67bdb42c45d8b24dc8f464ccd80883R115
The removal of `Logger` on existing SDKs will be done later once I have finished reports changes.
@sivachandran @maknahar @sudarshan-reddy

Reply:
>Sorry for the late reply. Can we not use the logs when debugging ad-hoc? Seems like if we add unit tests, we might not have the need to log debugs all that much at all. I see no harm in having it, but it just adds sirupsen/logrus as an extra package.

@sudarshan-reddy Whole point of using this interface is to make it logging package independent. It does not add any logging package level dependency.